src/utils/parsers.rs

```diff
@@ -20,9 +22,43 @@ pub fn parse_schema_source(loc: &str) -> Result<SchemaSource> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct GraphRef {
+    pub name: String,
+    pub variant: String,
+}
+
+/// NOTE: THIS IS A TEMPORARY SOLUTION. IN THE FUTURE, ALL GRAPH ID PARSING
+/// WILL HAPPEN IN THE BACKEND TO KEEP EVERYTHING CONSISTENT. THIS IS AN
+/// INCOMPLETE PLACEHOLDER, AND MAY NOT COVER EVERY SINGLE VALID USE CASE
+pub fn parse_graph_ref(graph_id: &str) -> Result<GraphRef> {
+    let pattern = Regex::new(r"^[a-zA-Z][a-zA-Z0-9_-]{0,63}$").unwrap();
+    let variant_pattern = Regex::new(r"^([a-zA-Z][a-zA-Z0-9_-]{0,63})@(.{0,63})$").unwrap();
+
+    let valid_graph_name_only = pattern.is_match(graph_id);
+    let valid_graph_with_variant = variant_pattern.is_match(graph_id);
+
+    if valid_graph_name_only {
+        Ok(GraphRef {
+            name: String::from(graph_id),
+            variant: String::from("current"),
```

Comment:
nit: use `to_string` instead of `String::from` - this is purely a style thing though as they are equivalent (i just happen to prefer `.to_string()`